### PR TITLE
feat(blocks): Ability to expand blocks input 

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Adds the ability to expand blocks input

### How to test it?

On clicking expand button at the bottom-right of blocks input, editor should open in fullscreen with content aligned in the middle of the modal as per [design](https://www.figma.com/file/MWPKHNT98LLWHGRp3MyWqu/Blocks?type=design&node-id=1333-5165&mode=design&t=Lu6K0vRvzYk99OKD-0).

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  );
+};
+
+const ExpandWrapper = styled(Flex)`
+  // Background with 10% opacity
+  background: ${({ theme }) =>
+    `${theme.colors.neutral900}${Math.floor(0.1 * 255)
+      .toString(16)
+      .padStart(2, '0')}`};
+`;
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx:105]: - could you move this above the main component? We usually put styled components near the top
- you can use #rrggbbaa [like in the design system](https://github.com/strapi/design-system/blob/main/packages/strapi-design-system/src/ModalLayout/ModalLayout.tsx#L17):

```
background: ${({ theme }) => `${theme.colors.neutral800}1F`};
```

it also matches the design: neutral800 with 20% opacity

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  ariaDescriptionId,
+}: EditorLayoutProps) => {
+  const { formatMessage } = useIntl();
+
+  useLockScroll({ lockScroll: isExpandMode });
+
+  if (isExpandMode) {
+    return (
+      <Portal role="dialog" aria-modal={false}>
+        <FocusTrap onEscape={onCollapse} skipAutoFocus>
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx:41]: I assume you checked that it wasn't possible to use any of the modal components from the DS here?

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/tests/BlocksInput.test.tsx [hunk 3]
```diff
@@ -90,4 +93,23 @@ describe('BlocksInput', () => {
 
     expect(screen.getByRole('img')).toBeInTheDocument();
   });
+
+  it('should open editor expand portal when clicking on expand button', async () => {
+    const { queryByTestId } = setup({ value: blocksData });
+
+    expect(queryByTestId('blocks-expand')).not.toBeInTheDocument();
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/tests/BlocksInput.test.tsx:100]: Why do you need the test id? Can't you check for the absence of the close button, or of the absence of the element with role modal?

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/tests/BlocksInput.test.tsx [hunk 4]
```diff
@@ -90,4 +93,23 @@ describe('BlocksInput', () => {
...
+    await user.click(screen.getByText('Expand'));
+
+    expect(screen.getByTestId('blocks-expand')).toBeInTheDocument();
+  });
+
+  it('should close editor expand portal when clicking on collapse button', async () => {
+    const { queryByTestId } = setup({ value: blocksData });
+
+    await user.click(screen.getByText('Expand'));
+    await user.click(screen.getByText('Collapse'));
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/tests/BlocksInput.test.tsx:111]: I would add a check in between to make sure the modal actually gets opened

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx (hunk 1 above)
Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx:105]: thanks

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/tests/BlocksInput.test.tsx (hunk 3 above)
Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/tests/BlocksInput.test.tsx:100]: indeed, role modal I thought of but its not very precise to test, checking collapse button is more feasible 👍

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx (hunk 2 above)
Comment (remidej) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/EditorLayout.tsx]: I'm guessing ModalLayout?

**Review by remidej — COMMENTED** [review]
The changes LGTM, can you just try using ModalLayout as mentioned in the thread above?

**Comment by madhurisandbhor** [comment]
> The changes LGTM, can you just try using ModalLayout as mentioned in the thread above?

I tried using ModalLayout but that doesn't help. Using FocusTrap is useful even to restore back focus on collapse.

**Review by remidej — DISMISSED** [review]
I tested, it looks good. I just misses the tooltip on the expand/collapse button. If you have time today that would be ideal, otherwise we can merge and release the tooltip in a patch

**Review by remidej — APPROVED** [review]

**Pull request merged**